src/java/org/apache/cassandra/net/OutboundTcpConnection.java

```diff
@@ -255,6 +259,24 @@ else if (socket != null || connect())
         }
     }
 
+    @VisibleForTesting
+    int clearQueueWithFailureCallback(int currentMessage, List<QueuedMessage> bufferedMessages, int bufferSize, BlockingQueue<QueuedMessage> queue) {
+        bufferedMessages.stream().skip(currentMessage).forEach(this::invokeFailureCallback);
```

Comment:
Are we not failing the current message as well? Considering we have skipped the `writeConnected` and entered the `else` statement above

Reply:
yeah, current message should be failed here as well, b/c currentMessage is this one's index, while we're skipping that many entries.

So currMessage 1 means we only skip index 0, and still fail index 1